tests/devices/test_default_qubit_torch.py

```diff
@@ -1260,12 +1342,12 @@ def circuit(x):
         res = circuit1(p_torch)
         res.backward()
 
-        assert np.allclose(res.detach().numpy(), circuit2(p).numpy(), atol=tol, rtol=0)
+        assert np.allclose(qml.math.to_numpy(res), circuit2(p).numpy(), atol=tol, rtol=0)
 
         p_grad = p_torch.grad
-        assert np.allclose(p_grad.detach().numpy(), qml.jacobian(circuit2)(p), atol=tol, rtol=0)
+        assert np.allclose(qml.math.to_numpy(p_grad), qml.jacobian(circuit2)(p), atol=tol, rtol=0)
```

Comment:
```suggestion
        assert qml.math.allclose(p_grad, qml.jacobian(circuit2)(p), atol=tol, rtol=0)
```

If we use `qml.math.allclose` we don't have to convert types and detach.

Reply:
Great catch! Thank you